src/main/java/org/parabot/core/ui/RandomUI.java

```diff
@@ -61,20 +68,31 @@ public void openFrame(ArrayList<String> randoms) {
 
     @Override
     public void actionPerformed(ActionEvent e) {
-        Context.getInstance().getRandomHandler().clearActiveRandoms();
-        if (checkBoxes != null && checkBoxes.size() > 0) {
-            for (JCheckBox checkBox : this.checkBoxes) {
-                if (checkBox.isSelected()) {
-                    for (Random r : Context.getInstance().getRandomHandler().getRandoms()) {
-                        if (r.getName().equalsIgnoreCase(checkBox.getText().toLowerCase())) {
-                            Core.verbose("Actived random '" + r.getName() + "'");
-                            Context.getInstance().getRandomHandler().setActive(r.getName());
+        org.parabot.api.output.Logger.debug("RandomUI", e.toString()+" | "+e.getActionCommand());
+        if (e.getActionCommand().startsWith("Run now: ")) {
+            org.parabot.api.output.Logger.debug("RandomUI", e.getActionCommand().substring(9));
+            for (Random r : Context.getInstance().getRandomHandler().getRandoms()) {
+                if (r.getName().equalsIgnoreCase(e.getActionCommand().substring(9))) {
```

Comment:
This substring isn't totally clear to me, could you elaborate this?

Reply:
Agreed basing the link to a random on text alone is weak. We could attach the linked randoms' class via usage of `clientProperty` which is just a small hashmap (key:random, value:class). Not exactly what it was designed for but small footprint and reduced code of creating a map of button:random ourselves.